modules/lead/product-base/staging.tf

```diff
@@ -126,3 +130,75 @@ resource "kubernetes_role" "ci_staging_role" {
     verbs      = ["*"]
   }
 }
+
+
+resource "vault_database_secret_backend_connection" "mongodb" {
+  backend = "mongodb"
+  name    = "${var.product_name}-staging"
+
+  allowed_roles = [
+    local.vault_mongodb_role
+  ]
+
+  data = {
+    username = "root"
+    password = random_password.mongodb_root_password.result
+  }
+
+  mongodb {
+    connection_url = "mongodb://{{username}}:{{password}}@mongodb.${var.product_name}-db.svc.cluster.local/admin"
+  }
+
+  depends_on = [
+    helm_release.mongodb
+  ]
+}
+
+resource "vault_database_secret_backend_role" "mongodb_role" {
+  backend = "mongodb"
+  creation_statements = [
+    jsonencode({
+      db = "admin"
+      roles = [
+        {
+          role = "readWrite"
+          db   = "database"
+        }
+      ]
+    })
+  ]
+
+  db_name = vault_database_secret_backend_connection.mongodb.name
+  name    = local.vault_mongodb_role
+
+  default_ttl = 60
+  max_ttl     = 60
```

Comment:
these ttl values were picked in order to demo this functionality, but in reality they should not be this aggressive.  perhaps every 5 or 10 minutes would be better

Reply:
Increased vault token timeout to 10 minutes